webknossos-datastore/app/com/scalableminds/webknossos/datastore/explore/ExploreRemoteLayerService.scala

```diff
@@ -106,26 +107,95 @@ class ExploreRemoteLayerService @Inject()(dataVaultService: DataVaultService,
-            exploreRemoteLayersForRemotePath(remotePath, credentialId, reportMutable, remainingExplorers)
-          case Empty =>
-            reportMutable += s"Error when reading $remotePath as ${currentExplorer.name}: Empty"
-            exploreRemoteLayersForRemotePath(remotePath, credentialId, reportMutable, remainingExplorers)
+    remotePathsWithDepth match {
+      case Nil =>
+        Fox.empty
+      case (path, searchDepth) :: remainingPaths =>
+        if (searchDepth > MAX_RECURSIVE_SEARCH_DEPTH) Fox.empty
+        else {
+          explorePathsWithAllExplorersAndGetFirstMatch(path, explorers, credentialId, reportMutable).futureBox.flatMap(
+            explorationResultOfPath =>
+              handleExploreResultOfPath(explorationResultOfPath,
+                                        path,
+                                        searchDepth,
+                                        remainingPaths,
+                                        credentialId,
+                                        explorers,
+                                        reportMutable))
         }
     }
 
+  private def explorePathsWithAllExplorersAndGetFirstMatch(path: VaultPath,
+                                                           explorers: List[RemoteLayerExplorer],
+                                                           credentialId: Option[String],
+                                                           reportMutable: ListBuffer[String])(
+      implicit ec: ExecutionContext): Fox[List[(DataLayerWithMagLocators, VoxelSize)]] =
+    Fox
+      .sequence(explorers.map { explorer =>
+        {
+          explorer
+            .explore(path, credentialId)
+            .futureBox
+            .flatMap {
+              handleExploreResult(_, explorer, path, reportMutable)
+            }
+            .toFox
+        }
+      })
+      .map(explorationResults => Fox.firstSuccess(explorationResults.map(_.toFox)))
+      .toFox
+      .flatten
+
+  private def handleExploreResult(explorationResult: Box[List[(DataLayerWithMagLocators, VoxelSize)]],
+                                  explorer: RemoteLayerExplorer,
+                                  path: VaultPath,
+                                  reportMutable: ListBuffer[String])(
+      implicit ec: ExecutionContext): Fox[List[(DataLayerWithMagLocators, VoxelSize)]] = explorationResult match {
+    case Full(layersWithVoxelSizes) =>
+      reportMutable += s"Found ${layersWithVoxelSizes.length} ${explorer.name} layers at $path."
+      Fox.successful(layersWithVoxelSizes)
+    case f: Failure =>
+      reportMutable += s"Error when reading $path as ${explorer.name}: ${Fox.failureChainAsString(f)}"
+      Fox.empty
+    case Empty =>
+      reportMutable += s"Error when reading $path as ${explorer.name}: Empty"
+      Fox.empty
+  }
+
+  // In case a data layers were found, explore sibling directories for more layers, else continue recursive exploration.
```

Comment:
I thought we’re not doing siblings anymore?

Reply:
Sorry, I simply forgot to remove this comment. Originally it was above the method doing the sibling exploration